streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBWindowStore.java

```diff
@@ -73,12 +71,12 @@ public void put(final K key, final V value) {
     public void put(final K key, final V value, final long windowStartTimestamp) {
         maybeUpdateSeqnumForDups();
 
-        bytesStore.put(WindowKeySchema.toStoreKeyBinary(key, windowStartTimestamp, seqnum, serdes), serdes.rawValue(value));
+        wrappedStore().put(WindowKeySchema.toStoreKeyBinary(key, windowStartTimestamp, seqnum, serdes), serdes.rawValue(value));
```

Comment:
This is ugly -- we should have access to the wrapped store directly via inheritance. (also below)

Reply:
I'm not sure what to say... It seems like good design, like encapsulation of state and referential transparency, should trump aesthetics about whether you prefer your references with or without parentheses?

We have many places in this code base where violating the encapsulation of state leads to poor maintainability and dangerous bugs. This is partly what we're trying to address in this PR.

If you like the name "wrapped" better than "wrappedStore", I can shorten it.

Reply:
Why does calling a method instead of accessing the variable prevent any bugs? It returns a reference anyway (and the variable is `final`). Compare my other comment about my mental model about inheritance for this case.

Reply:
Let's move this discussion to the other comment. It contains the explanation.